pkgs/by-name/xi/xiasl/package.nix

```
@@ -0,0 +1,80 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeDesktopItem
+, copyDesktopItems
+, qt5
+, libsForQt5
+, imagemagick
+, acpica-tools
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "xiasl";
+  version = "1.1.67";
+
+  src = fetchFromGitHub {
+    owner = "ic005k";
+    repo = "Xiasl";
+    rev = finalAttrs.version;
+    hash = "sha256-6RJeNLgeFchCjYLcwKgGtV0dLyHO7gYpIpo/3BBh/wM=";
+  };
+
+  buildInputs = [
+    qt5.qtbase
+  ];
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    qt5.wrapQtAppsHook
+    qt5.qmake
+    imagemagick
+  ];
+
+  postPatch = ''
+    rm libqscintilla2_qt5.so
+    ln -s '${lib.getLib libsForQt5.qscintilla}/lib/libqscintilla2_qt5.so' .
+
+    substituteInPlace Xiasl.pro \
+      --replace-warn '/opt/$''${TARGET}' "$out"
+
+    substituteInPlace *.cpp \
+      --replace-quiet 'appInfo.filePath() + "/iasl"' '"${acpica-tools}/bin/iasl"' \
+      --replace-quiet 'appInfo.filePath() + "/acpidump"' '"${acpica-tools}/bin/acpidump"' \
+  '';
+
+  postInstall = ''
+    icon_size="$(identify -format "%wx%h" icon.png)"
+    mkdir -p "$out/share/icons/hicolor/$icon_size/apps"
+    cp icon.png "$out/share/icons/hicolor/$icon_size/apps/xiasl.png"
```

```suggestion
    install -Dm644 icon.png "$out/share/pixmaps/xiasl.png"
```

These two lines are unnecessary.
```
icon_size="$(identify -format "%wx%h" icon.png)"
mkdir -p "$out/share/icons/hicolor/$icon_size/apps"
```

---

Done, thanks!